sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1492,16 +1520,31 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
     } else {
       Seq.empty
     }
-    val name = getFunctionMultiparts(func.functionName)
```

Comment:
I think we can use `PlanWithIdentifierClause` here
```
withIdentClause(ctx.identifierReference, name => {
  if (name.length > 1) ...
  val tvf = UnresolvedTableValuedFunction(name, func.expression.asScala.map(expression).toSeq)
  val tvfAliases = if (aliases.nonEmpty) UnresolvedTVFAliases(name, tvf, aliases) else tvf
  tvfAliases.optionalMap(func.tableAlias.strictIdentifier)(aliasPlan)
})
```

Reply:
It's not that simple :-(

Reply:
This may help
`  def withFuncIdentClause(ctx: FunctionNameContext, builder: Seq[String]
    => LogicalPlan): LogicalPlan = {
    if (ctx.expression != null) {
      PlanWithUnresolvedIdentifier(withOrigin(ctx) { expression(ctx.expression) }, builder)
    } else {
      builder.apply(getFunctionMultiparts(ctx))
    }
  }
`